gamemodes/terrortown/gamemode/client/cl_vskin/vgui/dcombobox_ttt2.lua

```diff
@@ -252,47 +319,142 @@ end
 ---
 -- @realm client
 function PANEL:CloseMenu()
-	if not IsValid(self.menu) then return end
+	if IsValid(self.menu) then
+		self.menu:Remove()
+		self.menu = nil
+	end
+end
 
-	self.menu:Remove()
+---
+-- @warning this doesnt set the displayed text like before, but the value and selects an option
+-- @param string/number value should be indexable
+-- @param bool ignoreConVar To avoid endless loops, separated setting of convars and UI values
+-- @realm client
+function PANEL:SetValue(value, ignoreConVar)
+	self:ChooseOption(value, nil, ignoreConVar)
+end
+
+---
+-- @param Panel menu to set the value of
+-- @param string conVar name of the convar
+local function AddConVarChangeCallback(menu, conVar)
+	local function OnConVarChangeCallback(conVarName, oldValue, newValue)
+		if not IsValid(menu) then
+			cvars.RemoveChangeCallback(conVarName, "TTT2F1MenuConVarChangeCallback")
+
+			return
+		end
+
+		menu:SetValue(newValue, true)
+	end
+
+	cvars.AddChangeCallback(conVar, OnConVarChangeCallback, "TTT2F1MenuConVarChangeCallback")
```

Comment:
"TTT2F1MenuConVarChangeCallback" is not unique enough, eg. when having the same variable multiple times for whatever reason, only one callback is registered. Maybe add a unique number or identifier.

Reply:
Is this the GMod provided function? You can leave the identifier away if you don't plan on removing the callback again

Reply:
I need the callback identifier as I remove the function, when the dropdown is inactive.